flow/testing/mock_layer.cc

```diff
@@ -16,12 +16,12 @@ MockLayer::MockLayer(SkPath path,
                      bool fake_reads_surface,
                      bool fake_opacity_compatible,
                      bool fake_has_texture_layer)
-    : fake_paint_path_(path),
-      fake_paint_(paint),
-      fake_has_platform_view_(fake_has_platform_view),
-      fake_reads_surface_(fake_reads_surface),
-      fake_opacity_compatible_(fake_opacity_compatible),
-      fake_has_texture_layer_(fake_has_texture_layer) {}
+    : fake_paint_path_(path), fake_paint_(paint) {
```

Comment:
This implementation misses the point completely. The thing we're trying to fix is those 4 bool arguments to the constructor. Internal implementation doesn't really matter, but we have a whole bunch of calls to this constructor in testing code that all have up to 4 "false, true, false, false" arguments that don't indicate what they are true/false-ing. I want the constructor to be something that makes sense to a casual reader, such as:

```
MockLayer layer(path, paint, HAS_PLATFORM_VIEW | IS_OPACITY_COMPATIBLE);
```

Then we can look at that line of code and understand *EXACTLY* what kind of layer is being constructed for the test.

Another solution would be something with chaining setters like:

```
MockLayer layer = MockLayer(path, paint)
    .set_fake_has_platform_view(true)
    .set_fake_opacity_compatibility(true);
```